mods/tnt/init.lua

```diff
@@ -415,6 +416,9 @@ minetest.register_node("tnt:gunpowder", {
 	on_blast = function(pos, intensity)
 		tnt.burn(pos, "tnt:gunpowder")
 	end,
+	on_burn = function(pos)
+		minetest.set_node(pos, {name = "tnt:gunpowder_burning"})
```

Comment:
it's as you said, in many cases the gunpowder gets lit in the edges (unless the node below the gunpowder is destroyed first). I think this effect is fine, and I guess it makes sense.

I've made the change that you suggested, not relying on `tnt.burn`. Actually, I'm not sure if `tnt.burn` is needed at all (other than for backwards compatibility).  Maybe if #1340 gets merged, the `tnt.burn` could just call the `on_ignite` of the node, so the gunpowder could be used to ignite anything a flint-and-steel can ignite.